### PR TITLE
Skip dimension of size 1 in contiguous checks.

### PR DESCRIPTION
**Pull request by MatthieuToulemont**
Since PyTorch 1.13 dimension of size 1 have normalised strides in PyTorch which fail here when using DLPack. This was done to conform the torch stride representation and the one from numpy. Unfortunately this means we are stuck with PyTorch 1.12.0 in python models.

Here is the [related issue on the server side](https://github.com/triton-inference-server/server/issues/6102). 

I haven't found any contributing guidelines as to what format Pull Requests should take for the `python_backend` will be happy to apply any changes you deem necessary. 

**Description of the Torch Issue since PyTorch 1.13.0**
PyTorch updated the way they handle strides for dimensions of size 1. 

In Pytorch 1.12 : 

```
import torch
from torch.utils.dlpack import to_dlpack, from_dlpack

x = torch.randn((1, 1, 768,768)).cuda()

y = from_dlpack(to_dlpack(x))

print(x.stride()) # (589824, 589824, 768, 1)
print(y.stride()) # (589824, 589824, 768, 1)
```

In PyTorch `>=` 1.13: 

```
import torch
from torch.utils.dlpack import to_dlpack, from_dlpack

x = torch.randn((1, 1, 768,768)).cuda()

y = from_dlpack(to_dlpack(x))

print(x.stride()) # (589824, 589824, 768, 1)
print(y.stride()) # (1, 1, 768, 1)
```

Closes: https://github.com/triton-inference-server/server/issues/6102

### CODE REVIEW

**Comment by MatthieuToulemont** [comment]
@Tabrizian Have you had some time to look at this ? 

This would enable using PyTorch >= 1.13 in python models in Triton which would enable torch compile too

**Comment by Tabrizian** [comment]
Thanks for fixing this issue. This PR looks good to me. Could you please sign the CLA as instructed here: https://github.com/triton-inference-server/server/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla

**Review by Tabrizian — APPROVED** [review]

**Comment by Tabrizian** [comment]
Looks like there is a formatting issue: https://github.com/triton-inference-server/python_backend/actions/runs/5659827069/job/15605216705?pr=281

Could you please format the PR using the `pre-commit` instructions here: https://github.com/triton-inference-server/server/blob/main/CONTRIBUTING.md#coding-convention

**Comment by Tabrizian** [comment]
thanks, I can confirm that we've received the CLA. I'm running the CI now and will merge if the CI looks green.

**Comment by MatthieuToulemont** [comment]
@Tabrizian I forgot to ask, do you know when this will be shipped in the Triton container ?

**Comment by Tabrizian** [comment]
This should be included in the 23.09 release.

**Comment by MatthieuToulemont** [comment]
Perfect, thank you very much !!
